### ISSUE TITLE
clang-scan-deps ignores warning supressions

### ISSUE THREAD

**Issue by anarthal** [llvm/llvm-project#185394]
With the following file:

```cpp
module;

export module mylib;

#pragma clang diagnostic push
#pragma clang diagnostic ignored "-Winclude-angled-in-module-purview"
#include <header.hpp> // header contents irrelevant
```

And the following build command:

```sh
#!/bin/bash

set -e

SRCDIR="$(cd "$(dirname "$0")" && pwd)"

clang-scan-deps-22 \
    -- \
    clang++-22 \
    -I"$SRCDIR" \
    -stdlib=libc++ \
    -g \
    -std=c++23 \
    -x c++ \
    "$SRCDIR/mylib.cpp" \
    -c -o mylib.cpp.o 
```

The pragma seems to be ignored:

```
Diagnostics while scanning dependencies for '[redacted]/modules-pg2/mylib.cpp':
[redacted]/modules-pg2/mylib.cpp:8:10: warning: '#include <filename>' attaches the declarations to the named module 'mylib', which is not usually intended; consider moving that directive before the module declaration
mylib.cpp.o: [redacted]/modules-pg2/mylib.cpp \
  [redacted]/modules-pg2/header.hpp
```

The compiler does not emit the diagnostic. It is only the scanner. This creates a cluttered log when legitimately including files in the purview to create module bindings.

**Comment by Bigcheese** [comment]
My general thoughts on this is that we should just suppress most warnings in the scanner. The normal compilation step will emit them. Only warnings that indicate scanning may have gone wrong should be emitted.

**Comment by ChuanqiXu9** [comment]
> My general thoughts on this is that we should just suppress most warnings in the scanner. The normal compilation step will emit them. Only warnings that indicate scanning may have gone wrong should be emitted.

Sounds good to me. Would you guys like to do this? I see you guys are active on the scanners recently. If you don't have a plan, I'll try to find time to make it.
